### PR TITLE
DOC: contribution guidelines

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
This PR improves our contribution guidelines to make it easier to begin contributing to ImageIO. They also add documentation of the v3 plugin API (the main motivation for this rewrite).

@almarklein Would you like to split this into two PRs? One for the contribution page improvement, and the other for the v3 plugin docs

### CODE REVIEW

File: docs/development/index.rst [hunk 1]
```diff
@@ -1,20 +1,238 @@
...
+no fixed format) and add a reference to the issue in which we discussed them.
+From there, we will help you by reviewing your changes, discussing them and
+eventually merging your work into ImageIO.
+
+Thank you for contributing!
+
+
+
+Code Contributions
+------------------
```
Comment (almarklein) [inline docs/development/index.rst:173]: Maybe move this up? I at least expect code contributions to be more "popular" than devops.

**Comment by almarklein** [comment]
> @almarklein Would you like to split this into two PRs? One for the contribution page improvement, and the other for the v3 plugin docs

This only contains contributor docs so far, right? Yes, keeping them separate keeps the pr more self-contained, easier to review, and faster to merge ;)

**Comment by almarklein** [comment]
lGTM so far

**Comment by FirefoxMetzger** [comment]
@almarklein I think this concludes the contributing doc rewrite. I will try to give it a second read in a few days (one tends to see new things when re-reading a document after a few days), but otherwise I am happy with the PR.

**Review by almarklein — APPROVED** [review]

**Pull request merged**